test/e2e/multihoming.go

```diff
@@ -444,18 +538,248 @@ var _ = Describe("Multi Homing", feature.MultiHoming, func() {
+				podConfiguration{ // client is host-networked
+					name:         clientPodName,
+					hostNetwork:  true,
+					isPrivileged: true,
+				},
+				podConfiguration{ // server on localnet
+					attachments: []nadapi.NetworkSelectionElement{{
+						Name: secondaryNetworkName,
+					}},
+					containerCmd:        httpServerContainerCmd(port),
+					name:                podName,
+					ipRequestFromSubnet: fromHostSubnet,
+				},
+				false, // collocated on different nodes
+				Label("STORY", "SDN-5345"),
+			),
+			Entry(
+				"can be reached by a host-networked pod on the same node, when the localnet uses an IP in the host subnet",
+				networkAttachmentConfigParams{
+					name:     secondaryNetworkName,
+					topology: "localnet",
+				},
+				podConfiguration{ // client is host-networked
+					name:         clientPodName,
+					hostNetwork:  true,
+					isPrivileged: true,
+				},
+				podConfiguration{ // server on localnet
+					attachments: []nadapi.NetworkSelectionElement{{
+						Name: secondaryNetworkName,
+					}},
+					containerCmd:        httpServerContainerCmd(port),
+					name:                podName,
+					ipRequestFromSubnet: fromHostSubnet,
+				},
+				true, // collocated on the same node
+				Label("STORY", "SDN-5345"),
+			),
+			// The second setup we test configures: a localnet that uses a VLAN, an external router
+			// that acts as gateway for the localnet pod for traffic destined to the host network.
+			// We implement the external router as an external container, where we create a VLAN interface
+			// on top of eth0 and assign to it an IP in the subnet in use by the localnet.
+			//
+			//                            +-----------------------+
+			//                            |     Kubernetes Node   |
+			//                            |       ovn-worker2     |
+			//                            |                       |
+			//    podA (10.244.1.10/24)---+-------[ br-int ]------+--- podL (172.20.0.4/16, net1)
+			//    (default net)           |           |           |     (localnet, VLAN 10)
+			//                            |       [ br-ex ]       |
+			//                            |        172.18.0.2     |
+			//                            +-----------|-----------+
+			//                                        |
+			//                                  host network
+			//                                  172.18.0.0/16
+			//                                        |
+			//                           +------------------------+
+			//                           |     external router    |
+			//                           |                        |
+			//                           |  eth0: 172.18.x.x      |
+			//                           |  eth0.10: 172.20.0.55  |
+			//                           +------------------------+
+			//
+			// Packet path (ping podA → podL):
+			//   podA (10.244.1.10)
+			//     → br-int
+			//       → br-ex (172.18.0.2, SNAT to node IP)
+			//         → eth0 (external router, 172.18.0.55)
```

Comment:
Is the IP here supposed to be `172.20.0.55` instead ? I'm a bit confused by the .55 within the 172.18.0.0 subnet.

Reply:
I've removed that, it's not important for the diagram